internal/client.go

```diff
@@ -1003,6 +1024,26 @@ func DialCloudOperationsClient(ctx context.Context, options CloudOperationsClien
 	}, nil
 }
 
+// PrepareUpdateWorkflowOperation returns an UpdateWorkflowOperation that can be used to perform Update-with-Start,
+// or an error in case the update operation is invalid.
+func PrepareUpdateWorkflowOperation(options UpdateWorkflowOptions) (*UpdateWorkflowOperation, error) {
+	input, err := createUpdateWorkflowInput(options)
+	if err != nil {
+		return nil, err
+	}
+	return &UpdateWorkflowOperation{input: input}, nil
+}
+
+// Get blocks on the outcome of the update.
+func (op *UpdateWorkflowOperation) Get(ctx context.Context, valuePtr interface{}) error {
+	if op.WorkflowUpdateHandle == nil {
+		return fmt.Errorf("update operation has not been executed")
+	}
+	return op.WorkflowUpdateHandle.Get(ctx, valuePtr)
+}
```

Comment:
I got here after some trial and error. Options I've considered:

**(1) `Get(ctx context.Context, valuePtr interface{}) error`** (as written here)

When the operation embeds `WorkflowUpdateHandle`, an extra step to obtain the handle is removed entirely. While this is the most convenient option, it also means that there is a risk the user invokes the handle's functions _before_ executing the operation (however, if `Get` is invoked too early, it returns the appropriate error).

```go
var result any
if err := upd.Get(context.Background(), &result); err != nil {
    return nil, err 
}
``` 

**(2) `Get(run WorkflowRun) (WorkflowUpdateHandle, error)`**

The safest but most verbose option. The `run` parameter is used to ensure the function is invoked _after_ `ExecuteWorkflow`. But it requires an extra error check that is _only_ there to prevent a mistake by the user, not to surface an actual runtime error.

```go
updHandle, err := upd.Get(context.Background(), run, &result)
if err != nil {
    return nil, err
}

var result any
if err := updHandle.GetResult(context.Background() &result); err != nil {
    return nil, err 
}
``` 

**(2.1) `GetResult(ctx context.Context, run WorkflowRun, valuePtr interface{}) error`**

Combined with (2), adding an additional shortcut to get the update outcome directly compresses two calls into one. This is based on the assumption that in most cases, users care about the result more than anything else from the `WorkflowUpdateHandle` (but they could still obtain it, if needed).

```go
var result any
if err := upd.GetResult(context.Background(), run, &result); err != nil {
    return nil, err 
}
``` 

**(3) `Handle(run WorkflowRun) WorkflowUpdateHandle`**

Removes the error check from (2). If the user passes in `nil`, it would panic (akin to a NPE).

```go
var result any
if err := upd.Handle(run).Get(context.Background(), &result); err != nil {
    return nil, err 
}
``` 

---

Essentially it comes down to safety vs convenience. Since all users would benefit from less verbose code, I think the question is how likely the user would make the mistake to request the result too early.